staging/src/k8s.io/apiserver/pkg/server/options/encryptionconfig/config.go

```diff
@@ -271,9 +274,20 @@ func (h *kmsv2PluginProbe) check(ctx context.Context) error {
 
 	h.lastResponse = &kmsPluginHealthzResponse{err: nil, received: time.Now()}
 	h.ttl = kmsPluginHealthzPositiveTTL
+	h.keyID.Store(&p.KeyID)
```

Comment:
I think you should set this before `isKMSv2ProviderHealthy` as long as its non-empty because even if a plugin is reporting failing healthz does not mean it cannot update its key ID.